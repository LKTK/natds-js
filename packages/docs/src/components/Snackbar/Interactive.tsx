import * as React from 'react';
import {
  Snackbar,
  SnackbarContent,
  Typography,
  Icon,
  Button,
  IconButton,
} from '@naturacosmeticos/natds-web';
import { makeStyles } from '@material-ui/core/styles';
import { select, number } from '@storybook/addon-knobs';

const colors: any = {
  default: 'default',
  error: 'error',
  warning: 'warning',
  success: 'success',
  info: 'info',
};

const rangeOptions: any = {
  range: true,
  min: 500,
  max: 10000,
  step: 500,
};

const verticalAnchor: any = {
  top: 'top',
  bottom: 'bottom',
};

const horizontalAnchor: any = {
  left: 'left',
  right: 'right',
  center: 'center',
};

export const Interactive = () => {
  const [isOpen, toggleOpen] = React.useState(false);

  const handleClose = () => {
    toggleOpen(false);
  };

  const useStyles = makeStyles({
    message: {
      display: 'flex',
      alignItems: 'center',
    },
    messageIcon: {
      marginRight: '10px',
    },
  });

  const classes = useStyles();

  return (
    <>
      <Button onClick={() => toggleOpen(true)} variant="contained">
        Click me to toggle the Snackbar
      </Button>
      <Snackbar
        open={isOpen}
        anchorOrigin={{
          vertical: select(
            'anchorOrigin.vertical',
            verticalAnchor,
            verticalAnchor.bottom
          ),
          horizontal: select(
            'anchorOrigin.horizontal',
            horizontalAnchor,
            horizontalAnchor.left
          ),
        }}
        autoHideDuration={number('autoHideDuration', 2000, rangeOptions)}
        onClose={handleClose}
      >
        <SnackbarContent
          color={select('color', colors, colors.default)}
          action={
            <IconButton color="inherit" onClick={handleClose}>
              <Icon name="filled-action-cancel" size="tiny" />
            </IconButton>
          }
          message={
            <span className={classes.message}>
              <span className={classes.messageIcon}>
                <Icon name="filled-brand-naturarosacea" size="small" />
              </span>
              <Typography variant="body2">This is a message</Typography>
            </span>
          }
        />
      </Snackbar>
    </>
  );
};
