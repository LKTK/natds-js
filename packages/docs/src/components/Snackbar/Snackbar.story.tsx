import withJest from '@decorators/jest/jest';
import withContainer from '@decorators/container/withContainer';

import { Snackbar } from '@naturacosmeticos/natds-web';
export { Interactive } from './Interactive';

import SnackbarDocs from './Snackbar.docs.mdx';

export default {
  title: 'Components|Snackbar',
  component: Snackbar,
  decorators: [withJest(), withContainer],
  parameters: {
    jestImportPath: 'web',
    jest: ['Snackbar', 'SnackbarContent'],
    theme: {
      context: 'web',
    },
    docs: {
      page: SnackbarDocs,
    },
  },
};
