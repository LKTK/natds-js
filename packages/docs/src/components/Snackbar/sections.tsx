import React from 'react';
import {
  SnackbarContent,
  Typography,
  Icon,
  IconButton,
  Button,
} from '@naturacosmeticos/natds-web';
import { makeStyles } from '@material-ui/core/styles';

const SnackbarContentWrapper = (props: any) => {
  const useStyles = makeStyles({
    message: {
      display: 'flex',
      alignItems: 'center',
    },
    messageIcon: {
      marginRight: '10px',
    },
  });

  const classes = useStyles();
  return (
    <SnackbarContent
      {...props}
      action={
        <IconButton color="inherit">
          <Icon name="filled-action-cancel" size="tiny" />
        </IconButton>
      }
      message={
        <span className={classes.message}>
          <span className={classes.messageIcon}>
            <Icon name="filled-brand-naturarosacea" size="small" />
          </span>
          <Typography variant="body2">This is a message</Typography>
        </span>
      }
    />
  );
};

const SnackbarButtonExample = (props: any) => {
  const useStyles = makeStyles({
    message: {
      display: 'flex',
      alignItems: 'center',
    },
    messageIcon: {
      marginRight: '10px',
    },
  });

  const classes = useStyles();
  return (
    <SnackbarContent
      {...props}
      action={props.action}
      message={
        <span className={classes.message}>
          <span className={classes.messageIcon}>
            <Icon name="filled-brand-naturarosacea" size="small" />
          </span>
          {props.message}
        </span>
      }
    />
  );
};

const SnackbarMultiLineText = (props: any) => {
  const useStyles = makeStyles({
    message: {
      display: 'flex',
      alignItems: 'center',
    },
    root: {
      maxWidth: '288px',
    },
  });

  const classes = useStyles();
  return (
    <SnackbarContent
      {...props}
      className={classes.root}
      message={<span className={classes.message}>{props.message}</span>}
    />
  );
};

export const colors = [
  {
    title: 'default',
    component: <SnackbarContentWrapper color="default" />,
  },
  {
    title: 'error',
    component: <SnackbarContentWrapper color="error" />,
  },
  {
    title: 'success',
    component: <SnackbarContentWrapper color="success" />,
  },
  {
    title: 'warning',
    component: <SnackbarContentWrapper color="warning" />,
  },
  {
    title: 'info',
    component: <SnackbarContentWrapper color="info" />,
  },
];

export const examples = [
  {
    title: 'with button',
    component: (
      <SnackbarButtonExample
        action={
          <Button variant="text" color="inherit">
            button
          </Button>
        }
        message={<Typography variant="body2">This is a message</Typography>}
      />
    ),
  },
  {
    title: 'multiline text',
    component: (
      <SnackbarMultiLineText
        message={
          <Typography variant="body2">
            This is a really long message to see how it works on multiline
          </Typography>
        }
      />
    ),
  },
];
