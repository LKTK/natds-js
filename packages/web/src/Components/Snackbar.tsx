import * as React from 'react';
import MaterialSnackbar, { SnackbarProps as ISnackbarProps } from '@material-ui/core/Snackbar';
import { withTheme} from '@material-ui/core/styles';
export { SnackbarProps as ISnackbarProps } from '@material-ui/core/Snackbar';

export const Snackbar:React.FC<ISnackbarProps> = (props: ISnackbarProps) => {
  return <MaterialSnackbar {...props} />;
};

export default withTheme(Snackbar);
