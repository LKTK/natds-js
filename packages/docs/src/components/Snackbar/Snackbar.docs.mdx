import { TextField, Provider, themes } from '@naturacosmeticos/natds-web';
import ComponentWithLabel from '../../../addons/ComponentWithLabel';
import { colors } from './sections';

# Snackbar

---
---
---
---

## Colors

<Provider theme={themes.natura.light}>
  <ComponentWithLabel componentList={colors} />
</Provider>

---
---

## API

```jsx highlight-line="2"
import React from 'react';
import {
  Snackbar,
  SnackbarContent,
  Typography,
  Icon,
  IconButton,
  Button
} from '@naturacosmeticos/natds-web';
import { makeStyles } from '@material-ui/core/styles';


// This is optional and can be made with any stylesheet method that you prefer
const useStyles = makeStyles({
  message: {
    display: 'flex',
    alignItems: 'center'
  },
  messageIcon: {
    marginRight: '10px'
  }
});
const classes = useStyles();

const [isOpen, setIsOpen] = React.useState(false);

const handleClose = () => setIsOpen(false);

const myApp = () => (
  <Button onClick={() => setIsOpen(true)}>Click me to open the snackbar</Button>

  <Snackbar open={isOpen} autoHide>
    <SnackbarContent {...props}
      message={
        <span className={classes.message}>
          <span className={classes.messageIcon}>
            <Icon name="outlined-navigation-close" size="small"/>
          </span>
          <Typography>This is a message</Typography>
        </span>
      }
      action={
        <IconButton color="inherit" onClick={handleClose}>
          <Icon name="filled-alert-cancel" size="tiny" />
        </IconButton>
      }
    />
  </Snackbar>
);

```
For more advanced usages, check [Material UI Snackbar docs](https://material-ui.com/components/snackbars/)

---

---

## SnackbarContent Props
---
---

|Name|Type|Default|Description|
|--- |--- |--- |--- |
|color|'default \| 'error' \| 'success' \| 'warning' \| 'info' | 'default' | The color scheme to be applied to the content based on the theme
|action|node||The action to display.|
|classes|object||Override or extend the styles applied to the component. See CSS API below for more details.|
|message|node||The message to display.|
|role|'alert' \| 'alertdialog'|'alert'|The role of the SnackbarContent. If the Snackbar requires focus to be closed, the alertdialog role should be used instead.|


## Snackbar Props
---
---

|Name|Type|Default|Description|
|--- |--- |--- |--- |
|action|node||The action to display.|
|anchorOrigin|{ horizontal: 'left'\| 'center'\| 'right', vertical: 'top'\| 'bottom' }|{ vertical: 'bottom', horizontal: 'center' }|The anchor of the Snackbar.|
|autoHideDuration|number||The number of milliseconds to wait before automatically calling the onClose function. onClose should then set the state of the open prop to hide the Snackbar. This behavior is disabled by default with the null value.|
|children|element||Replace the SnackbarContent component.|
|classes|object||Override or extend the styles applied to the component. See CSS API below for more details.|
|ClickAwayListenerProps|object||Props applied to the ClickAwayListener element.|
|ContentProps|object||Props applied to the SnackbarContent element.|
|disableWindowBlurListener|bool|false|If true, the autoHideDuration timer will expire even if the window is not focused.|
|key|any||When displaying multiple consecutive Snackbars from a parent rendering a single `<Snackbar/>`, add the key prop to ensure independent treatment of each message. e.g. `<Snackbar key={message} />`, otherwise, the message may update-in-place and features such as autoHideDuration may be canceled.|
|message|node||The message to display.|
|onClose|func||Callback fired when the component requests to be closed. Typically onClose is used to set state in the parent component, which is used to control the Snackbar open prop. The reason parameter can optionally be used to control the response to onClose, for example ignoring clickaway.Signature:function(event: object, reason: string) => voidevent: The event source of the callback.reason: Can be:"timeout" (autoHideDuration expired) or: "clickaway".|
|onEnter|func||Callback fired before the transition is entering.|
|onEntered|func||Callback fired when the transition has entered.|
|onEntering|func||Callback fired when the transition is entering.|
|onExit|func||Callback fired before the transition is exiting.|
|onExited|func||Callback fired when the transition has exited.|
|onExiting|func||Callback fired when the transition is exiting.|
|open|bool||If true, Snackbar is open.|
|resumeHideDuration|number||The number of milliseconds to wait before dismissing after user interaction. If autoHideDuration prop isn't specified, it does nothing. If autoHideDuration prop is specified but resumeHideDuration isn't, we default to autoHideDuration / 2 ms.|
|TransitionComponent|elementType|Grow|The component used for the transition.|
|transitionDuration|number| { enter?: number, exit?: number }|{  enter: duration.enteringScreen,  exit: duration.leavingScreen,}|The duration for the transition, in milliseconds. You may specify a single timeout for all transitions, or individually with an object.|
|TransitionProps|object||Props applied to the Transition element.|

