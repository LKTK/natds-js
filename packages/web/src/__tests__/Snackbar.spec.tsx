import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Snackbar from '../Components/Snackbar';

describe('Snackbar Component', () => {
  describe('Rendering', () => {
    test('should match to snapshot - Snackbar default', () => {
      const wrapper = renderer.create(<Snackbar open={false} />);

      expect(wrapper.toJSON()).toMatchSnapshot('Snackbar default snapshot');
    });
  });
});
