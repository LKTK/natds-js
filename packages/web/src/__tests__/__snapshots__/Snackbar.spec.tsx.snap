// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar Component Rendering should match to snapshot - Snackbar default: Snackbar default snapshot 1`] = `null`;
